voila/paths.py

```diff
@@ -19,66 +20,78 @@
     # relative to the package directory (first entry, meaning with highest precedence)
-    search_directories = []
-    if DEV_MODE:
-        search_directories.append(os.path.abspath(os.path.join(ROOT, '..', 'share', 'jupyter', 'voila', 'templates')))
-    search_directories.extend(jupyter_path('voila', 'templates'))
+    if root_dirs is None:
+        root_dirs = []
+        if DEV_MODE:
+            root_dirs.append(os.path.abspath(os.path.join(ROOT, '..', 'share', 'jupyter', 'templates')))
+        root_dirs.extend(jupyter_path('templates'))
 
     found_at_least_one = False
-    for search_directory in search_directories:
-        template_directory = os.path.join(search_directory, template_name)
-        if os.path.exists(template_directory):
-            found_at_least_one = True
-            conf = {}
-            conf_file = os.path.join(template_directory, 'conf.json')
-            if os.path.exists(conf_file):
-                with open(conf_file) as f:
-                    conf = json.load(f)
+    paths = []
+    full_paths = []  # only used for error reporting
 
-            # For templates that are not named 'default', we assume the default base_template is 'default'
-            # that means that even the default template could have a base_template when explicitly given.
-            if template_name != 'default' or 'base_template' in conf:
-                collect_template_paths(
-                    nbconvert_template_paths,
-                    static_paths,
-                    tornado_template_paths,
-                    conf.get('base_template', 'default'))
+    for root_dir in root_dirs:
+        if include_root_paths:
+            # we include root_dir for when we want to be very explicit, e.g.
+            # {% extends 'nbconvert/classic/base.html' %}
+            paths.append(root_dir)
+        for app_name in app_names:
+            app_dir = os.path.join(root_dir, app_name)
+            if include_root_paths:
+                # we include app_dir for when we want to be explicit, but less than root_dir, e.g.
+                # {% extends 'classic/base.html' %}
+                paths.append(app_dir)
+            full_paths.append(app_dir)
 
-            extra_nbconvert_path = os.path.join(template_directory, 'nbconvert_templates')
-            nbconvert_template_paths.insert(0, extra_nbconvert_path)
+            template_dir = os.path.join(app_dir, template_name)
+            if os.path.exists(template_dir):
+                # if we are intested in a subdirectory instead
+                if subdir:
+                    paths.append(os.path.join(template_dir, subdir))
+                else:
+                    paths.append(template_dir)
 
-            extra_static_path = os.path.join(template_directory, 'static')
-            static_paths.insert(0, extra_static_path)
+                found_at_least_one = True
+                conf_file = os.path.join(template_dir, 'conf.json')
```

Comment:
Is there still a need for `conf.json`? If so maybe we can provide an example?

Reply:
Yes, to say which is the base template, e.g. https://github.com/QuantStack/voila-vuetify/blob/master/share/jupyter/voila/templates/vuetify-default/conf.json

Reply:
:+1: